File: files/200-299/287/CH-287.R
library(tidyverse)
library(readxl)

path = "files/200-299/287/CH-287 Transformation.xlsx"
input = read_excel(path, range = "B2:E6")
test  = read_excel(path, range = "I2:K15")

result = input %>%
  pivot_longer(cols = everything(), names_to = "Date", values_to = "Value", values_drop_na = T) %>%
  mutate(Date = str_sub(Date, 1, 11)) %>%
  separate(Value, into = c("Product", "Value"), sep = "-") %>%
  mutate(Value = as.numeric(Value),
         Date = anytime::anydate(Date) %>% as.POSIXct(format = "%d/%m/%Y", tz = "UTC")) 

all.equal(result, test, check.attributes = F)

File: files/200-299/287/CH-287.py
import pandas as pd

path = "200-299/287/CH-287 Transformation.xlsx"

input = pd.read_excel(path, usecols="B:E", skiprows=1, nrows=4)
test = pd.read_excel(path, usecols="I:K", skiprows=1, nrows=13)

result = (
    input.melt(var_name="Date", value_name="Value")
    .dropna(subset=["Value"])
    .assign(
        Date=lambda df: pd.to_datetime(df["Date"].astype(str).str[:11], format="%d/%b/%Y"),
        Product=lambda df: df["Value"].str.split("-", n=1).str[0],
        Value=lambda df: pd.to_numeric(df["Value"].str.split("-", n=1).str[1], errors="coerce"),
    )
    .sort_values(["Product", "Date"])
    .reset_index(drop=True)[["Date", "Product", "Value"]]
)

print(result.equals(test)) # True